### PR TITLE
Rebase auto random counter after restore

### LINKED ISSUE #241: BR full restore won't rebase `auto_random` counter
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
BR didn't rebase auto_random after restore

    1.  set tidb config `auto_random` to true.
    2. `create table t(a int key auto_random); insert into t values (), (), ();`
    3. `br backup table --db t --table t --pd xxx -s xxx`
    4. `drop table t`
    5. `br restore table --db t --table t --pd xxx -s xxx` 
    6. `insert into t values (),(),();`

before backup:
```
MySQL [t]> select a & b'0000011111111111111111111111111' from t.t;
+----------------------------------------+
| a & b'0000011111111111111111111111111' |
+----------------------------------------+
|                                      1 |
|                                      2 |
|                                      3 |
+----------------------------------------+
```
2. What did you expect to see?

after restore:
```
select a & b'0000011111111111111111111111111' from t.t;
+----------------------------------------+
| a & b'0000011111111111111111111111111' |
+----------------------------------------+
|                                      4 |
|                                      5 |
|                                      6 |
|                                      1 |
|                                      2 |
|                                      3 |
+----------------------------------------+
```

3. What did you see instead?

after restore:
```
select a & b'0000011111111111111111111111111' from t.t;
+----------------------------------------+
| a & b'0000011111111111111111111111111' |
+----------------------------------------+
|                                      1 |
|                                      2 |
|                                      3 |
|                                      1 |
|                                      2 |
|                                      3 |
+----------------------------------------+
```

4. What version of BR and TiDB/TiKV/PD are you using?
master

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
This PR Fix #241 
with this PR we can rebase auto_random increment part counter.

### What is changed and how it works?
1. backup with auto random allocator to get nextGlobalRandID, if table enabled auto_random
2. set nextGlobalRandID to tableInfo
3. rebase auto_random counter if table enabled

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: pkg/restore/db.go [hunk 1]
```diff
@@ -119,6 +120,26 @@ func (db *DB) CreateTable(ctx context.Context, table *utils.Table) error {
 			zap.Stringer("table", table.Info.Name),
 			zap.Error(err))
 	}
+	if table.Info.PKIsHandle && table.Info.ContainsAutoRandomBits() {
+		// this table has auto random id, we need rebase it
+
+		// we can't merge two alter query, because
+		// it will cause Error: [ddl:8200]Unsupported multi schema change
+		alterAutoRandIDSQL := fmt.Sprintf(
+			"alter table %s.%s auto_random = %d",
```
Comment (kennytm) [inline pkg/restore/db.go:129]:
```suggestion
			"alter table %s.%s auto_random_base = %d",
```

File: tests/br_incompatible_tidb_config/run.sh [hunk 2]
```diff
@@ -111,5 +111,27 @@ run_br --pd $PD_ADDR restore db --db "$DB" -s "local://$TEST_DIR/$DB$INCREMENTAL
 
 run_sql "drop schema $DB;"
 
+# test auto random issue https://github.com/pingcap/br/issues/241
+TABLE="t4"
+run_sql "create schema $DB;"
+run_sql "create table $DB.$TABLE(a int key auto_random);"
```
Comment (kennytm) [inline tests/br_incompatible_tidb_config/run.sh:117]:
```suggestion
run_sql "create table $DB.$TABLE(a int key auto_random(5));"
```

File: pkg/backup/client.go [hunk 3]
```diff
@@ -226,6 +227,20 @@ func BuildBackupRangeAndSchema(
 				return nil, nil, errors.Trace(err)
 			}
 			tableInfo.AutoIncID = globalAutoID
+
+			if tableInfo.PKIsHandle && tableInfo.ContainsAutoRandomBits() {
```
Comment (GregoryIan) [inline pkg/backup/client.go:231]: must it to meet both requirements?

Comment (3pointer) [reply pkg/backup/client.go:231]: yes, auto_random must be assigned on PKIsHandle table.

**Pull request merged**